src/components/auth-code/auth-code-service.ts

```diff
@@ -7,25 +7,48 @@ import {
   Http,
 } from "../../utils/http";
 import { AuthCodeResponse, AuthCodeServiceInterface } from "./types";
-import { getApiBaseUrl } from "../../config";
+import {
+  getApiBaseUrl,
+  getFrontendApiBaseUrl,
+  supportAuthOrchSplit,
+} from "../../config";
+import { AxiosResponse } from "axios";
 
 export function authCodeService(axios: Http = http): AuthCodeServiceInterface {
   const getAuthCode = async function (
     sessionId: string,
     clientSessionId: string,
     sourceIp: string,
-    persistentSessionId: string
+    persistentSessionId: string,
+    clientSession: UserSessionClient
   ): Promise<ApiResponseResult<AuthCodeResponse>> {
-    const response = await axios.client.get(
-      API_ENDPOINTS.AUTH_CODE,
-      getRequestConfig({
-        baseURL: getApiBaseUrl(),
-        sessionId: sessionId,
-        clientSessionId: clientSessionId,
-        sourceIp: sourceIp,
-        persistentSessionId: persistentSessionId,
-      })
-    );
+    const baseUrl = supportAuthOrchSplit()
+      ? getFrontendApiBaseUrl()
+      : getApiBaseUrl();
+    const config = getRequestConfig({
+      baseURL: baseUrl,
+      sessionId: sessionId,
+      clientSessionId: clientSessionId,
+      sourceIp: sourceIp,
+      persistentSessionId: persistentSessionId,
+    });
+
+    let response: AxiosResponse;
+
+    if (supportAuthOrchSplit()) {
+      const body = {
+        claim: clientSession.claim,
```

Comment:
I think these will need to match the backend's request fields?:

https://github.com/alphagov/di-authentication-api/blob/main/frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/entity/AuthCodeRequest.java#L9

So:

- "redirect-uri"
- "state"
- "claims"

Reply:
🤦‍♀️ good spot